### ISSUE TITLE
New HomeKit Automations Not Executing based on SS Sensor State Changes

### ISSUE THREAD

**Issue by mypupsrgr8** [nzapponi/homebridge-simplisafe3#63]
Hello,

I don’t know that this is an issue or if this is expected behavior.  I couldn’t find a way to ask a question without creating an issue.

I have created a couple of new automations, one that turns my ecobee 3 thermostats off when a door opens and executes a scene to resume the schedule on them when the door is closed.  I tested the automations and they worked and are enabled, however, they aren’t’ the executing when the door sensor changes state.  DomI need to restart Homebridge on my Mac Mini every time an automation that requires a Homebridge-enabled accessory is created and/or modified, or do I just need to tell the associated plugin(s) to reload?  If so, how do I only restart/reload a specific Homebridge plugin without stopping and starting Homebridge?

Best Regards,

Tee Allen

**Comment by nikonratm** [comment]
Hi Tee, we are currently working on a couple of issues that may be related to this, see #59, #56, #61 .

The short answer is while there doesnt seem to be a "problem" with the plugin there is some combination of HomeKit's impatience and the slowness of SS servers that we are trying to resolve. In the meantime, I will say that some of my automations dont fire for reasons that definitely dont have to do with this plugin (or Homebridge). Among other things I have had to restart my hub (e.g. Apple TV), _reset_ my hub, delete and re-create automations, etc. And to answer your question you cant restart a single HB plugin, only the "server" itself (which is why some people create multiple HB installations).

In short, try the above and if no luck—hang in there, we're working on it (I say we but @nzapponi does the lion's share, obviously)!

**Comment by mypupsrgr8** [comment]
Hi,

Thanks for the prompt reply.  I am still running the tvOS 13.3.1 beta on my Apple TV I believe unless it automatically updated to 13.3.1 when it officially was released a couple of days ago.  My time-based non-SimpliSafe automations still seem to be working, so maybe my current issue is related to the things you are trying to resolve.  I did reduce the sensor polling from 15 to 5 seconds and that improved things.  Up until this issue, I only had to restart Homebridge once because none of the SimpliSafe sensor states were being updated.  I was afraid it might have been because I took the polling time interval too low, but  it has been a couple of days now and the state changes are still showing up in the Home app, but the automations just aren’t firing.

Best Regards,

Ted Allen

Sent from my iPad

On Jan 29, 2020, at 10:27 PM, Michael Shamoon <notifications@github.com> wrote:

﻿

Hi Tee, we are currently working on a couple of issues that may be related to this, see #59<https://github.com/nzapponi/homebridge-simplisafe3/issues/59>, #56<https://github.com/nzapponi/homebridge-simplisafe3/issues/56>, #61<https://github.com/nzapponi/homebridge-simplisafe3/pull/61> .

The short answer is while there doesnt seem to be a "problem" with the plugin there is some combination of HomeKit's impatience and the slowness of SS servers that we are trying to resolve. In the meantime, I will say that some of my automations dont fire for reasons that definitely dont have to do with this plugin (or Homebridge). Among other things I have had to restart my hub (e.g. Apple TV), reset my hub, delete and re-create automations, etc. And to answer your question you cant restart a single HB plugin, only the "server" itself (which is why some people create multiple HB installations).

In short, try the above and if no luck—hang in there, we're working on it (I say we but @nzapponi<https://github.com/nzapponi> does the lion's share, obviously)!

**Comment by nzapponi** [comment]
Should be fixed in release v1.4.3 going out in the next few days

**Comment by nikonratm** [comment]
Unfortunately I dont think the above release will quite fix this issue (though it definitely helps with other stuff). At least it didn't for me but since HK / HB problems are often quite varied in their cause maybe it will help @mypupsrgr8 .

But I am hopeful the PR #67 I submitted will help (see #66). Also, @mypupsrgr8 there is an issue with HomeKit "turn off after" automations, see the bottom of that issue. One suggestion is to try https://github.com/nitaybz/homebridge-delay-switch which I'm going to try.

**Comment by nikonratm** [comment]
Update: I _think_ the [delay-switch](https://github.com/nitaybz/homebridge-delay-switch) work-around addresses the HomeKit "turn off after..." issue, see https://github.com/nzapponi/homebridge-simplisafe3/issues/66#issuecomment-580941160

**Comment by nzapponi** [comment]
Probably, but its any case it's not something that should be implemented in this plugin, right?

**Comment by nikonratm** [comment]
> Probably, but its any case it's not something that should be implemented in this plugin, right?

Oh yea agreed, not something for this plugin, I just posted here for the next poor soul who arrives here on their frustrating quest to fix non-working HomeKit automations :)

**Comment by nzapponi** [comment]
See #66 as well
